docs/src/main/asciidoc/async-message-passing.adoc

```diff
@@ -0,0 +1,239 @@
+include::./attributes.adoc[]
+= {project-name} - Asynchronous messages between beans
+
+{project-name} allows different beans to interact using asynchronous messages, enforcing loose-coupling.
+The messages are sent to _virtual addresses_.
+It offers 3 types of delivery mechanism:
+
+- point-to-point - send the message, one consumer receives it. If several consumers listen to the address, a round robin is applied;
+- publish/subscribe - publish a message, all the consumers listening to the address are receiving the message;
+- request/reply - send the message and expect a response. The receiver can respond to the message in an asynchronous-fashion
+
+All these delivery mechanism are non-blocking, and are providing one of the fundamental brick to build reactive applications.
+
+== Installing
+
+This mechanism uses the Vert.x EventBus, so you need to enable the `vertx` extension to use this feature.
+If you are creating a new project, set the `extensions` parameter are follows:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=vertx-quickstart \
+    -Dextensions="vert.x"
+----
+
+If you have an already created project, the `vertx` extension can be added to an existing {project-name} project with
+the `add-extension` command:
+
+[source]
+----
+mvn quarkus:add-extension -Dextensions="vertx"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-vertx</artifactId>
+</dependency>
+----
+
+== Consuming events
+
+To consume events, use the `io.quarkus.vertx.ConsumeEvent` annotation:
+
+[source, java]
+----
+package org.acme;
+
+import io.quarkus.vertx.ConsumeEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class NameService {
+
+    @ConsumeEvent                           // <1>
+    public String consume(String name) {    // <2>
```

Comment:
I think a warning should be added indicating the message will be processed on an event loop, and the method should not use blocking APIs.

Reply:
Or we simply delay the documentation and revisit the implementation so that events are consumed on the regular xnio/undertows tasks thread pool.

Reply:
We are adding a `blocking` attribute (pending PR)

Reply:
But for now (as I don't want to wait for the release to publish these docs), I agree it must be mentioned.

Reply:
Done!